datastores/buffer.go

```diff
@@ -32,7 +32,7 @@ func BufferTemp(datastore config.DatastoreConfig, contents io.ReadCloser) (strin
 		logrus.Warnf("Datastore %s does not have a valid temporary path configured. This will lead to increased memory usage.", datastore.Id)
 		target = &bytes.Buffer{}
 	} else {
-		err = os.Mkdir(fpath, os.ModeDir)
+		err = os.Mkdir(fpath, os.ModeDir | 0700)
```

Comment:
```suggestion
		err = os.Mkdir(fpath, os.ModeDir | 0600)
```

I don't think we need execute bits

Reply:
We do, accessing files inside a directory requires the directory to have execute permissions.